src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImportOptionsCsv.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * DataImportOptionsCsv.java
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+package org.rstudio.studio.client.workbench.views.environment.dataimport;
+
+import com.google.gwt.json.client.JSONBoolean;
+import com.google.gwt.json.client.JSONNumber;
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.json.client.JSONString;
+
+public class DataImportOptionsCsv extends DataImportOptions
+{
+   public DataImportOptionsCsv(String dataName,
+                               Character delimiter,
+                               String quotes,
+                               Boolean escapeBackslash,
+                               Boolean escapeDouble,
+                               Boolean columnNames,
+                               Boolean trimSpaces,
+                               String locale,
+                               String na,
+                               String comments,
+                               int skip)
+   {
+      setDataName(dataName);
+      delimiter_ = delimiter;
+      quotes_ = quotes;
+      escapeBackslash_ = escapeBackslash;
+      escapeDouble_ = escapeDouble;
+      columnNames_ = columnNames;
+      trimSpaces_ = trimSpaces;
+      locale_ = locale;
+      na_ = na;
+      comments_ = comments;
+      skip_ = skip;
+   }
+   
+   public Character getDelimiter()
+   {
+      return delimiter_;
+   }
+   
+   public String getQuotes()
+   {
+      return quotes_;
+   }
+   
+   public Boolean getEscapeBackslash()
+   {
+      return escapeBackslash_;
+   }
+   
+   public Boolean getEscapeDouble()
+   {
+      return escapeDouble_;
+   }
+   
+   public Boolean getColumnNames()
+   {
+      return columnNames_;
+   }
+   
+   public Boolean getTrimSpaces()
+   {
+      return trimSpaces_;
+   }
+   
+   public String getLocale()
+   {
+      return locale_;
+   }
+   
+   public String getNa()
+   {
+      return na_;
+   }
+   
+   public String getComments()
+   {
+      return comments_;
+   }
+   
+   public int getSkip()
+   {
+      return skip_;
+   }
+   
+   @Override
+   public JSONObject toJSONObject()
```

Comment:
Similar to JJs comment, I'm implementing these class hierarchy in native JS to avoid all the overhead.